### PR TITLE
test images: Adds image labels by default

### LINKED ISSUE #99648: Add metadata to images built from test/e2e/images
from https://github.com/kubernetes/kubernetes/pull/99631#issuecomment-788945533

Specify metadata during build vs. hardcoding in Dockerfile to avoid duplication/drift (since we have multiple for linux, windows, etc.)

Specify as ARG in Dockerfile(s) and then can pass in via `--build-arg`

Could include metadata such as:
- git_url
- git_commit
- etc.

/assign @claudiubelu @mkumatag 
Can you flesh this out a bit more for me? I can update the description with more of a proposed spec.  Or if nothing else, here's a tracking issue for following up on the linked comment.
/milestone v1.21
/priority important-soon
I'm assuming this is something you plan on getting done this release, but please feel free to kick this out if not

/wg k8s-infra
/sig testing
/area test
/area artifacts
(side note, I feel like it's time to namespace `area/test`, I want to tag this with something like `area/test/images`, but gonna use `area/artifacts` for now)

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind cleanup

/sig testing

#### What this PR does / why we need it:

Adds the "image_version" and "commit_id" to the image labels.

#### Which issue(s) this PR fixes:

Fixes #99648

#### Special notes for your reviewer:

Image build with this PR:

```
docker inspect claudiubelu/busybox:1.29-1
[
    {
        "Id": "sha256:25cb596ea358107320b19aa0735da2359a6bbb3be612511b18ab18825a2a3bd2",
        "RepoTags": [
            "claudiubelu/busybox:1.29-1"
        ],
        "RepoDigests": [
            "claudiubelu/busybox@sha256:a6506eba8d8fe4651c10c0d55cab601fe22900784badd395325c834123922ceb"
        ],
        "Parent": "",
        "Comment": "",
        "Created": "2018-12-26T08:20:42.831353376Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "sh"
            ],
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "commit_id": "f8cdcc10562",
                "image_version": "1.29-1"
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 1154361,
        "VirtualSize": 1154361,
        "GraphDriver": {
            "Data": {
                "MergedDir": "/var/lib/docker/overlay2/474b4adbbd584a3b568cf74e7c18f778fc56168d61b3179aea8960179eb24512/merged",
                "UpperDir": "/var/lib/docker/overlay2/474b4adbbd584a3b568cf74e7c18f778fc56168d61b3179aea8960179eb24512/diff",
                "WorkDir": "/var/lib/docker/overlay2/474b4adbbd584a3b568cf74e7c18f778fc56168d61b3179aea8960179eb24512/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:23bc2b70b2014dec0ac22f27bb93e9babd08cdd6f1115d0c955b9ff22b382f5a"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
but this can wait for the branch to open for v1.23

File: test/images/image-util.sh [hunk 1]
```diff
@@ -182,7 +183,8 @@ build() {
 
     docker buildx build --progress=plain --no-cache --pull --output=type="${output_type}" --platform "${os_name}/${arch}" \
         --build-arg BASEIMAGE="${base_image}" --build-arg REGISTRY="${REGISTRY}" --build-arg OS_VERSION="${os_version}" \
-        -t "${REGISTRY}/${image}:${TAG}-${suffix}" -f "${dockerfile_name}" .
+        -t "${REGISTRY}/${image}:${TAG}-${suffix}" -f "${dockerfile_name}" \
+	--label "image_version=${TAG}" --label "commit_id=${GIT_COMMIT_ID}" .
```
Comment (mkumatag) [inline test/images/image-util.sh:187]: Wondering if we can add git_url as well which saves the location..

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Review by SataQiu — COMMENTED** [review]
/lgtm

File: test/images/image-util.sh (hunk 1 above)
Comment (claudiubelu) [reply test/images/image-util.sh]: Done

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Comment by claudiubelu** [comment]
Needs milestone.

**Comment by claudiubelu** [comment]
Needs milestone.

**Pull request merged**